pkg/vorteil/process.go

```diff
@@ -131,6 +134,44 @@ func shutdown(cmd, timeout int) {
 
 }
 
+func sendTerminateSignals() {
+	var wg sync.WaitGroup
+
+	// loop through program's terminate signals
+	for p, termSig := range terminateSignals {
+		wg.Add(1)
+		go func(np *program, sig syscall.Signal) {
+			defer wg.Done()
+
+			// if program is finished skip
+			if np.isDone {
+				return
+			}
+
+			// send terminate to program
+			if err := np.cmd.Process.Signal(sig); err != nil {
+				logError("could not send terminate signal program %v, error: %v", np.cmd.Process.Pid, err)
+			}
+
+			// Wait for process to be terminated...
+			np.cmd.Process.Wait()
```

Comment:
Afaik you are waiting no this process here and where you set p.isDone = true. One of those will "steal" the wait response. We should only use it once.